arrow-test/src/main/kotlin/arrow/test/eq/Instances.kt

```diff
@@ -0,0 +1,71 @@
+package arrow.test.eq
+
+import arrow.Kind
+import arrow.core.Either
+import arrow.core.Tuple2
+import arrow.core.extensions.either.eq.eq
+import arrow.core.extensions.tuple2.eq.eq
+import arrow.fx.ForIO
+import arrow.fx.IO
+import arrow.fx.extensions.io.applicative.applicative
+import arrow.fx.extensions.io.concurrent.waitFor
+import arrow.fx.fix
+import arrow.fx.typeclasses.Duration
+import arrow.fx.typeclasses.seconds
+import arrow.mtl.AccumT
+import arrow.mtl.AccumTPartialOf
+import arrow.mtl.Kleisli
+import arrow.mtl.KleisliPartialOf
+import arrow.mtl.StateT
+import arrow.mtl.StateTPartialOf
+import arrow.mtl.fix
+import arrow.mtl.run
+import arrow.typeclasses.Eq
+import arrow.typeclasses.EqK
+import arrow.typeclasses.Monad
+
+/**
+ * This is mainly useful for testing laws because those take a EqK without any further constraints, however when testing outside of those constraints
+ *  one should always generate random initial states to get a better coverage!
+ */
+fun <F, D> Kleisli.Companion.eqK(eqKF: EqK<F>, ctx: D): EqK<KleisliPartialOf<F, D>> = object : EqK<KleisliPartialOf<F, D>> {
+  override fun <A> Kind<KleisliPartialOf<F, D>, A>.eqK(other: Kind<KleisliPartialOf<F, D>, A>, EQ: Eq<A>): Boolean =
+    eqKF.liftEq(EQ).run { run(ctx).eqv(other.run(ctx)) }
+}
+
+/**
+ * This is mainly useful for testing laws because those take a EqK without any further constraints, however when testing outside of those constraints
+ *  one should always generate random initial states to get a better coverage!
+ */
+fun <F, S> StateT.Companion.eqK(EQKF: EqK<F>, EQS: Eq<S>, M: Monad<F>, s: S) = object : EqK<StateTPartialOf<F, S>> {
+  override fun <A> Kind<StateTPartialOf<F, S>, A>.eqK(other: Kind<StateTPartialOf<F, S>, A>, EQ: Eq<A>): Boolean =
+    (this.fix() to other.fix()).let {
+      val ls = it.first.run(s)
+      val rs = it.second.run(s)
+
+      EQKF.liftEq(Tuple2.eq(EQS, EQ)).run {
+        ls.eqv(rs)
+      }
+    }
+}
+
+fun IO.Companion.eqK(timeout: Duration = 60.seconds) = object : EqK<ForIO> {
```

Comment:
I found `60.seconds` rather long since a test should not last that long anyway or it should be the exception and not the baseline.

I would argue that a test shouldn't take longer than 1 second, especially in big test suite like we have. In the end I settled for `5.seconds` and that seemed to work fine for the current test suite.

For example, the MVar `concurrent take and put` test turned out to take longer than 5 seconds on CI sometimes, so we decided to reduce the test from `10_000` iterations to `5_000` since it's not testing for StackOverflow'ing anyway.